src/Form/BlockLinkField_Readonly.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace SilverStripe\ElementalBlocks\Form;
+
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\Forms\CheckboxField_Readonly;
+use SilverStripe\Forms\FieldList;
+use SilverStripe\Forms\ReadonlyField;
+use SilverStripe\Forms\TextField;
+use SilverStripe\Forms\ToggleCompositeField;
+use SilverStripe\Forms\TreeDropdownField;
+
+/**
+ * Readonly version of a {@link BlockLinkField} field, which displays the data fields as readonly text
+ * inputs and a checkbox for "target blank".
+ */
+class BlockLinkField_Readonly extends ReadonlyField
+{
+    public function Field($properties = [])
+    {
+        /** @var BlockLinkField $originalField */
+        $originalField = BlockLinkField::create('TempReadonly')->setValue($this->value);
+
+        $name = $this->getName();
+
+        $fields = FieldList::create();
+
+        $fields->push(
+            TreeDropdownField::create($name . '_PageID', null, SiteTree::class)
+                ->setTitle(_t('SilverStripe\\ElementalBlocks\\Form\\BlockLinkField.SelectPage', 'Select a page'))
+                ->setValue($originalField->getParsedValue()->PageID)
+                ->performReadonlyTransformation()
+        );
+
+        if ($originalField->getShowLinkText()) {
+            $fields->push(
+                $this->castedCopy(TextField::class)
```

Comment:
Should this & the description not be readonly or have a readonly transformation applied also?

Reply:
They're being set as readonly lower down this method, but I think you're right - it would be better to use the performReadonlyTransformation API consistently

Reply:
Sorry, just looked at it again. It's like this because it's casting itself as a TextField, which is a method that performReadonlyTransformation performs internally. The reason it's doing it this way is because we want to re-use as many of the internal properties on BlockLinkField as possible to inject into the casted copy, but we're specifying that it should be casted as a TextField for some of the data.

FYI this field is basically a composite JSON field, so this method is casting each key in the JSON data as an individual form field for readonly display.

Hope that makes sense?

Reply:
Yep, sounds legit.